src/components/menu/menu.pw.tsx

```diff
@@ -0,0 +1,2426 @@
+            callbackCount += 1;
+          }}
+          submenu="Menu Item One"
+        />
+      );
+
+      await page.keyboard.press("Tab");
+      await page.keyboard.press(key);
+      expect(callbackCount).toBe(1);
+    });
+  });
+
+  test(`should call onSubmenuClose callback when menu is closed`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <MenuComponentItems
+        onSubmenuClose={() => {
+          callbackCount += 1;
+        }}
+        submenu="Menu Item Three"
+      />
+    );
+
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("ArrowDown");
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onClose callback when Menu Fullscreen is closed`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await page.setViewportSize({ width: 1200, height: 800 });
+    await mount(
+      <MenuComponentFullScreen
+        onClose={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const item = menuItem(page).first();
+    await item.click();
+    const fullscreen = getComponent(page, "menu-fullscreen").first();
+    await waitForAnimationEnd(fullscreen);
+    const closeIcon = closeIconButton(page).first();
+    await closeIcon.click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should have correct keyboard navigation order when children of submenu update`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<MenuWithChildrenUpdating />);
+
+    const subMenu = submenu(page).first();
+    await subMenu.hover();
+    const subMenuBlock = submenuBlock(page).locator("li");
+    await expect(subMenuBlock).toHaveCount(4);
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("ArrowDown");
+    const focusedElement1 = page.locator("*:focus");
+    await expect(focusedElement1).toHaveText("Apple");
+    await page.keyboard.press("ArrowDown");
+    const focusedElement2 = page.locator("*:focus");
+    await expect(focusedElement2).toHaveText("Banana");
+    await page.keyboard.press("ArrowDown");
+    const focusedElement3 = page.locator("*:focus");
+    await expect(focusedElement3).toHaveText("Carrot");
+    await page.keyboard.press("ArrowDown");
+    const focusedElement4 = page.locator("*:focus");
+    await expect(focusedElement4).toHaveText("Broccoli");
+    await page.keyboard.press("ArrowUp");
+    const focusedElement5 = page.locator("*:focus");
+    await expect(focusedElement5).toHaveText("Carrot");
+    await page.keyboard.press("ArrowUp");
+    const focusedElement6 = page.locator("*:focus");
+    await expect(focusedElement6).toHaveText("Banana");
+    await page.keyboard.press("ArrowUp");
+    const focusedElement7 = page.locator("*:focus");
+    await expect(focusedElement7).toHaveText("Apple");
+  });
+});
+
+test.describe("Accessibility tests for Menu component", () => {
+  test(`should pass tests for default Menu`, async ({ mount, page }) => {
```

Comment:
```suggestion
  test(`should pass accessibility tests for default Menu`, async ({ mount, page }) => {
```
same as below for all the accessibility tests